TestTicTacToe/GameRulesTest.cs

```diff
@@ -0,0 +1,219 @@
+using TicTacToe;
+using Xunit;
+
+namespace TestTicTacToe
+{
+    public class GameRulesTest
+    {
+
+        [Fact]
+        public void Should_Return_False_When_3_In_A_Row_Horizontally_Of_Different_Players_Marks()
+        {
+            // Arrange
+            var gameLosingBoard = new[,]
+            {
+                {1, 2, 1},
+                {0, 0, 0},
+                {0, 0, 0}
+            };
+        
+            // Act
+            var hasWonGame = GameRulesHandler.HasWon(gameLosingBoard);
+        
+            // Assert
+            Assert.False(hasWonGame);
+        }
+        
+        
+        [Fact]
+        public void Should_Return_False_When_3_In_A_Row_Vertically_Of_Different_Players_Marks()
+        {
+            // Arrange 
+            var gameLosingBoard = new[,]
+            {
+                {0, 1, 0},
+                {0, 1, 0},
+                {0, 2, 0}
+            };       
+        
+            // Act
+            var hasWonGame = GameRulesHandler.HasWon(gameLosingBoard);
+        
+            // Assert
+            Assert.False(hasWonGame);
+        }
+
+        [Fact]
+        public void Should_Return_False_When_3_In_A_Row_Diagonally_As_Forward_Slash_Of_Different_Players_Marks()
+        {
+            // Arrange 
+            var gameLosingBoard = new[,]
+            {
+                {2, 0, 0},
+                {0, 1, 0},
+                {0, 0, 1}
+            };
+        
+        
+            // Act
+            var hasWonGame = GameRulesHandler.HasWon(gameLosingBoard);
+        
+            // Assert
+            Assert.False(hasWonGame);
+        }
+        
+        [Fact]
+        public void Should_Return_False_When_3_In_A_Row_Diagonally_As_Backward_Slash_Of_Different_Players_Marks()
```

Comment:
I wonder if all these tests for `Assert.False(hasWonGame)` can be combined into a `theory` or something else -- the scenario being tested is that it should return false when there is a 3 in a row but they aren't all one player, the only difference in each case is where the 3 in a row is

Reply:
same for checking for draw and checking for win, might be less verbose to / more readable to have 3 theories:
* not won scenario
* won scenario
* draw scenario

Reply:
Agree with Alvin - To keep test files small it is good to combine like scenarios in Theories